tests/unit/features/partitions/test_parts.py

```diff
@@ -0,0 +1,206 @@
+
+from textwrap import dedent
+
+import pytest
+
+from craft_parts import parts
+from craft_parts.parts import Part
+from tests.unit import test_parts
+
+
+class TestPartSpecs(test_parts.TestPartSpecs):
+    """Test part specification creation."""
+
+
+class TestPartData(test_parts.TestPartData):
+    """Test basic part creation and representation."""
+
+
+class TestPartOrdering(test_parts.TestPartOrdering):
+    """Test part ordering."""
+
+
+class TestPartUnmarshal(test_parts.TestPartUnmarshal):
+    """Verify data unmarshaling on part creation."""
+
+
+class TestPartHelpers(test_parts.TestPartHelpers):
+    """Test part-related helper functions."""
+
+
+class TestPartValidation(test_parts.TestPartValidation):
+    """Part validation considering plugin-specific attributes."""
+
+
+class TestPartPartitionUsage:
+    """Test usage of partitions in parts."""
+
+    @pytest.fixture
+    def partition_list(self):
+        """Return a list of partitions, 'default' and 'kernel'."""
+        return ["default", "kernel"]
+
+    @pytest.fixture
+    def valid_filesets(self):
+        """Return a list of valid filesets when the partition feature is enabled.
+
+        Assumes "default" and "partition" are valid partitions.
+        """
+        return [
+            "test",
+            "(default)",
+            "(default)/test",
+            "(default)/test/(otherdir)",
+            "(default)/test/(otherdir-123)",
+            "(default)/test/(default)",
+            "(default)/test/(kernel)",
+            "test(default)",
+            "test(default)/test",
+            "test/(default)/test",
+            "(kernel)",
+            "(kernel)/test",
+            "(kernel)/test/(otherdir)",
+            "(kernel)/test/(otherdir-123)",
+            "(kernel)/test/(default)",
+            "(kernel)/test/(kernel)",
+            "test/(kernel)",
+            "test(kernel)/test",
+        ]
+
+    @pytest.fixture
+    def invalid_filesets(self):
+        """Return a list of invalid filesets when the partition feature is enabled.
+
+        Assumes "default" and "partition" are the only valid partitions.
+        """
+        return [
+            "()",
+            "(foo)",
+            "(bar)/test",
+            "(baz)",
+            "(123)/test",
+            "(123)/test/(456)",
+            "(123)/test/(default)",
+            "(123)/test/(kernel)",
+        ]
+
+    def test_part_valid_partition_usage(self, valid_filesets, partition_list):
+        """Proper use of partitions in parts should not raise an error."""
+        part_data = {
+            "organize": {
+                f"test-{index}": item for index, item in enumerate(valid_filesets)
+            },
+            "stage": valid_filesets,
+            "prime": valid_filesets,
+        }
+
+        part_list = [Part("a", part_data), Part("b", part_data)]
+
+        assert not parts.validate_partition_usage(part_list, partition_list)
```

Comment:
```suggestion
        assert parts.validate_partition_usage(part_list, partition_list) is None
```

This is another nit, feel free to ignore: `not parts.validate_partition_usage(part_list, partition_list)` makes me think that the function is returning `False`, meaning that the validation failed, which is the opposite of the true meaning

Reply:
Sounds good.  That's a better way to assert that a function that doesn't return anything also does not raise an exception.